lib/preview-web/src/PreviewWeb.tsx

```diff
@@ -46,6 +47,16 @@ function focusInInput(event: Event) {
 }
 
 type MaybePromise<T> = Promise<T> | T;
+type PossibleRender<TFramework extends AnyFramework> =
```

Comment:
`Render` is the name of the interface. I  could change that to `BaseRender`?

Reply:
`IRender`? `BaseRender` sounds like it's providing some base functionality but it's just an interface. I wonder what's idiomatic TypeScript for this? I'm very familiar with this pattern from my OO days, but I haven't seen the analog to `PossibleRender` there.

Reply:
I can't really remember what it was like to code in an OO language but I think TS does stuff with type "elimination" that I don't recall from those days. Like if you have `const x: A | B` and then put `if (x not of type A)` in your code (that's pseudo code) TS will automatically cast it to `B` which is super handy.

Whereas if you have `const x: I` where `I` is an interface that `A` and `B` implement, it can't know that `x` isn't so other theoretical implementer, even if none exist.